src/workloads/execution/Densify.yml

```diff
@@ -0,0 +1,194 @@
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: NumericFullDensifyByPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "number", partitionByFields: ["partitionKey"], range: {bounds: "full", step: {^RandomInt: {min: 1, max: 11}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: NumericPartitionedDensifyByPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "number", partitionByFields: ["partitionKey"], range: {bounds: "partition", step: {^RandomInt: {min: 1, max: 11}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: NumericExplicitRangeDensify
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "number", range: {step: {^RandomInt: {min: 1, max: 11}}, bounds: [10000, 30000]}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: NumericExplicitRangeDensifyByPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "number", partitionByFields: ["partitionKey"], range: {step: {^RandomInt: {min: 1, max: 11}}, bounds: [10000, 30000]}}}]
+        cursor: {batchSize: *batchSize}
+    # Millisecond metrics.
+    - OperationMetricsName: MillisecondFullDensify
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "milliseconds", range: {bounds: "full", step: {^RandomInt: {min: 1, max: 11}}, unit: "millisecond"}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: MillisecondFullDensifyByPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "milliseconds", partitionByFields: ["partitionKey"], range: {bounds: "full", step: {^RandomInt: {min: 1, max: 11}}, unit: "millisecond"}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: MillisecondPartitionedDensifyByPartition
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "milliseconds", partitionByFields: ["partitionKey"], range: {bounds: "partition", unit: "millisecond", step: {^RandomInt: {min: 1, max: 11}}}}}]
+        cursor: {batchSize: *batchSize}
+    - OperationMetricsName: MillisecondExplicitRangeDensify
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline: [{$densify: {field: "milliseconds", range: {step: {^RandomInt: {min: 1, max: 11}}, unit: "millisecond", bounds: [{^RandomDate: {min: "2021-01-01T00:00:05.000Z", max: "2021-01-01T00:00:05.001Z"}}, {^RandomDate: {min: "2021-01-01T00:00:15.000Z", max: "2021-01-01T00:00:15.001Z"}}]}}}]
```

Comment:
The random min/max here are very small -- with this range may as well hardcode, or should come up with a better random bounds that guarantees max is above min.

Reply:
this range is hardcoded right now -- the only way to add a date type in a query is to use $^RandomDate with a min and max bound. I'll add a  comment explaining that, there's an open ticket to add in a better way to do this. I'll probably move the ranges so that they are actually random to add some variation, though.